### PR TITLE
Runs functional tests inside the container running Pulp

### PR DESCRIPTION
**Pull request by dkliban**
closes: #642

### CODE REVIEW

File: templates/github/.ci/ansible/Containerfile.j2.copy [hunk 1]
```diff
@@ -7,6 +7,12 @@ ADD {{ item.source }} {{ item.source }}
 {% endif %}
 {% endfor %}
 
+{%- for item in plugins -%}
+{%- if item.name == "pulp_container" -%}
+RUN dnf install -y podman
```
Comment (fao89) [inline templates/github/.ci/ansible/Containerfile.j2.copy:12]: why not having it by default?

File: templates/github/.ci/ansible/Containerfile.j2.copy [hunk 2]
```diff
@@ -7,6 +7,12 @@ ADD {{ item.source }} {{ item.source }}
 {% endif %}
 {% endfor %}
 
+{%- for item in plugins -%}
```
Comment (newswangerd) [inline templates/github/.ci/ansible/Containerfile.j2.copy:10]: @dkliban it seemed like (and I might be wrong about this) that the pip dependencies are installed each time CI is run. Would it be possible to install the basic pulpcore requirements [here](https://github.com/pulp/pulp-oci-images/blob/latest/pulp_ci_centos/Containerfile), and then run `pip install --upgrade` when the plugins are installed in CI?

A large chuck of the CI time is just downloading the pip packages during the container build. Would be nice to make that run faster.

File: templates/github/.github/workflows/scripts/script.sh.j2 [hunk 3]
```diff
@@ -56,27 +56,36 @@ if [[ "$TEST" == "plugin-from-pypi" ]]; then
   git checkout ${COMPONENT_VERSION} -- {{ plugin_name }}/tests/
 fi
 
+cat unittest_requirements.txt | cmd_stdin_prefix bash -c "cat > /tmp/unittest_requirements.txt"
+cat functest_requirements.txt | cmd_stdin_prefix bash -c "cat > /tmp/functest_requirements.txt"
+cmd_prefix pip3 install -r /tmp/unittest_requirements.txt
```
Comment (newswangerd) [inline templates/github/.github/workflows/scripts/script.sh.j2:61]: If I'm not mistaken, cmd_prefix just runs `docker exec`. Would it make sense to move these steps where you're just installing extra requirements in the container to the `Containerfile`?

File: templates/github/.ci/ansible/Containerfile.j2.copy (hunk 2 above)
Comment (dkliban) [reply templates/github/.ci/ansible/Containerfile.j2.copy:10]: That is a good idea.  I would like to implement it as a separate change from this.

File: templates/github/.ci/ansible/Containerfile.j2.copy (hunk 1 above)
Comment (dkliban) [reply templates/github/.ci/ansible/Containerfile.j2.copy:12]: Good idea. I opened a PR for this. https://github.com/pulp/pulp-oci-images/pull/203

File: templates/github/.github/workflows/scripts/script.sh.j2 (hunk 3 above)
Comment (dkliban) [reply templates/github/.github/workflows/scripts/script.sh.j2:61]: Yes, that would make sense. I'll make that change.

Comment (dkliban) [reply templates/github/.github/workflows/scripts/script.sh.j2:61]: I tried implementing this and it's not going to work without doing a lot of refactoring to how we build the container. The current Containerfile template doesn't care if the plugin is being installed from source (PR time) or from a prebuilt wheel (for release pipeline). When the wheel is being used, the functest_requirements.txt and unittest_requirements.txt are not available during the built.

**Review by fao89 — APPROVED** [review]
LGTM

**Pull request merged**